apps/sms/views/conversation/js/conversation.js

```diff
@@ -723,6 +725,45 @@ var ConversationView = {
     });
   },
 
+  handleComposer: function thui_handleComposer(composer, recipients) {
+
+    function generateMmsAttachmentsForComposer(slides, content) {
+      if (typeof content === 'string') {
+        slides.push({
+          content: new Blob([content], { type: 'text/plain' }),
+        });
+      } else {
+        slides.push({
+          content: content.blob,
+          location: content.name
+        });
+      }
+      return slides;
+    }
+
+    Compose.fromMessage({
+      subject: composer.subject,
+      type: composer.type,
+      body: composer.body,
+      attachments: composer.content
+        .reduce(generateMmsAttachmentsForComposer, [])
+    });
+    Compose.focus();
+
+    if (recipients) {
+      for (var i in recipients) {
+        Utils.getContactDisplayInfo(
```

Comment:
I don't know if you remember, but you asked me to factorize that with how we get contacts for handleActivity. Actually I think it's not exactly the same, so I kept it like that. The difference is that I never have a contact. Btw is there a way to get the current MozContact of a conversation? It would be handy to directly pass it to the Composer instead of doing what I do here.